### PR TITLE
go prev page after dismiss plaid modal

### PR DESCRIPTION
**Pull request by wenge8n**
### Details

When dismiss the plaid modal, it goes to home screen not previous screen.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7381

### Tests

1. Open app
2. Navigate to Settings > Payments > Add bank account > Plaid
3. Navigate Back

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open app
2. Navigate to Settings > Payments > Add bank account > Plaid
3. Navigate Back

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

[Video](https://imgur.com/a/U8j1kqE)

#### Desktop

[Video](https://imgur.com/a/VyOt4p5)

#### iOS

[Video](https://imgur.com/a/CcpUOS3)

#### Android

https://user-images.githubusercontent.com/36442590/151579780-5f6220da-8173-4026-8064-d50200fd404f.mp4

### CODE REVIEW

**Comment by rushatgabhane** [comment]
@wenge8n can you please add screenshots for all platforms?

**Comment by wenge8n** [comment]
@rushatgabhane I added videos for web/desktop/android/ios. I couldn't test on mobile web. I see only the loading spinner maybe due to the cors problem.

File: src/pages/AddPersonalBankAccountPage.js [hunk 1]
```diff
@@ -28,15 +28,15 @@ const AddPersonalBankAccountPage = props => (
         <KeyboardAvoidingView>
             <HeaderWithCloseButton
                 title={props.translate('bankAccount.addBankAccount')}
-                onCloseButtonPress={Navigation.dismissModal}
+                onCloseButtonPress={() => Navigation.goBack()}
```
Comment (rushatgabhane) [inline src/pages/AddPersonalBankAccountPage.js:31]: No need to create an anonymous func here. Just `onCloseButtonPress={Navigation.goBack}` is fine

File: src/pages/AddPersonalBankAccountPage.js [hunk 2]
```diff
@@ -28,15 +28,15 @@ const AddPersonalBankAccountPage = props => (
...
                 shouldShowBackButton
                 onBackButtonPress={() => Navigation.goBack()}
```
Comment (rushatgabhane) [inline src/pages/AddPersonalBankAccountPage.js:33]: same here

**Comment by rushatgabhane** [comment]
@wenge8n Did you follow any aditional steps to allow cors on web/desktop? Could you please help me with that?

**Comment by wenge8n** [comment]
When I test on desktop, there was no issue. When I test on web, I see cors issue. I installed cors extension on chrome and the issue disappeared.

**Comment by wenge8n** [comment]
@rushatgabhane of course I can write like `onCloseButtonPress={Navigation.goBack}`. But I created anonymous func because throughout the entire source code, this format is used.
Do you still want to make change?

**Comment by rushatgabhane** [comment]
>. But I created anonymous func because throughout the entire source code, this format is used.
Do you still want to make change?

Yes, we should avoid them to prevent re-renders. https://github.com/Expensify/App/blob/main/PERFORMANCE.md#react-performance-tips
> When I test on web, I see cors issue. I installed cors extension on chrome and the issue disappeared.

Thanks! [This chrome extension](https://chrome.google.com/webstore/detail/cors-unblock/lfhmikememgdcahcdlaciloancbhjino?hl=en) worked for me on web.

**Comment by wenge8n** [comment]
Ok I will make update.

> Thanks! This chrome extension worked for me on web.
Great!

**Comment by rushatgabhane** [comment]
@wenge8n Looks great!
On web, back from plaid page navigates to settings page instead of payments page. Is it possible for you to fix this?


https://user-images.githubusercontent.com/29673073/151710217-82824dfc-9052-44e0-9f59-e417a76b155a.mp4

**Comment by wenge8n** [comment]
@rushatgabhane I pushed the update.

**Comment by wenge8n** [comment]
@rushatgabhane 
> On web, back from plaid page navigates to settings page instead of payments page. Is it possible for you to fix this?

Do you have any suggestion to solve this? This is tricky problem.
This happens because by click the browser back button, we already navigated to the Payments screen. And from the onExit event, we navigate back once more so we go to Settings page.

**Comment by wenge8n** [comment]
> If it's due to the way Navigation works, then I think we shouldn't fix it. It's a different issue altogether.

Ok. then hope to solve this by another issue.

**Comment by rushatgabhane** [comment]
> Do you have any suggestion to solve this? This is tricky problem. This happens because by clicking the browser back button, we already navigated to the Payments screen. And from the onExit event, we navigate back once more so we go to Settings page.

@wenge8n Yep, `exit` event is fired after pressing browser back. Even if we fix it for the browser, we'll have to handle `cmd + [` on desktop too.

**Review by rushatgabhane — APPROVED** [review]
@chiragsalian overall LGTM. Tested on all platforms **except mWeb** because I can't get around the CORS issue. ngrok secure url isn't available externally.

We have [this bug](https://github.com/Expensify/App/pull/7452#issuecomment-1025193848), but I don't see any elegant way to fix it. So I recommend that we ignore it. Let me what you think :)

**Comment by chiragsalian** [comment]
Yeah, i think ignoring that bug on mWeb is fine for now. It doesn't feel worth it for that edge case right now.

**Review by chiragsalian — APPROVED** [review]

**Comment by rushatgabhane** [comment]
> i think ignoring that bug on mWeb 

oh no 😅 Sorry that I wasn't clear enough. [This edge case](https://github.com/Expensify/App/pull/7452#issuecomment-1025193848) is present on **web and desktop**. 

I think we should ignore it because we can't fix it in any clean way. The solution will be too much of a workaround, not worth fixing imo.

**Pull request merged**